4.next

skie opened the pull request.

codemile committed:
API fixes for CakePHP 3.8

codemile committed:
adds support for setting nulls when updating

codemile committed:
adds support deleting single records

codemile committed:
adds support for epilog to trigger queryAll instead of query

Zuluru committed:
Fix logging

Zuluru committed:
Improved error handling

Zuluru committed:
Boolean variables in queries need to be the literal strings 'true' or 'false', not quoted

skie committed:
step by step upgrade to 4.x

skie committed:
implements bulk operations

skie committed:
Merge branch 'cakephp-3-8' into 4.next

skie committed:
step by step upgrade to 4.x

skie committed:
implements Driver::getRestClient

skie committed:
Merge branch 'cakephp-3-8' into 4.next

skie committed:
step by step upgrade to 4.x

Zuluru commented on src/Database/Schema/SalesforceSchema.php:
```diff
@@ -45,7 +47,7 @@ public function listTables()
 
     /**
-     * Custom function that queries the datasource for the table list as SOQL doesn't accept
+     * Custom function that queries the datasource for the table list as SQL doesn't accept
```
This was actually SOQL on purpose; that's Salesforce's custom not-quite-standard SQL variant.

skie committed:
fix typo

skie committed:
update composer reqs

skie committed:
fix formating

steinkel committed:
update to getRepository

skie committed:
fix date marshaling

skie committed:
fix date marshaling

skie committed:
add safe date transforming

Zuluru committed:
Bulk operation tweaks

cnizzardini committed:
Check if `$config['connection']` is not empty

Check if `$config['connection']` is not empty to resolve bug:

```
Error: [Error] Call to a member function config() on null in src/Model/Table/SalesforcesTable.php on line 52
```

steinkel committed:
Merge pull request #1 from cnizzardini/patch-1

Check if `$config['connection']` is not empty in SalesforcesTable
